chore(oxc_linter): move `Message::from_tsgo_lint_diagnostic` behind the language_server feature flag

Sysix opened the pull request:
fixes the warning when running the command:

```
~/dev/oxc$ cargo test -p oxlint
warning: associated function `from_tsgo_lint_diagnostic` is never used
   --> crates/oxc_linter/src/tsgolint.rs:569:8
    |
567 | impl Message<'_> {
    | ---------------- associated function in this implementation
568 |     /// Converts a `TsGoLintDiagnostic` into a `Message` with possible fixes.
569 |     fn from_tsgo_lint_diagnostic(val: TsGoLintDiagnostic, source_text: &str) -> Self {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```

Boshen commented:
### Merge activity

* **Aug 29, 3:16 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
chore(oxc_linter): move `Message::from_tsgo_lint_diagnostic` behind the language_server feature flag (#13374)

fixes the warning when running the command:

```
~/dev/oxc$ cargo test -p oxlint
warning: associated function `from_tsgo_lint_diagnostic` is never used
   --> crates/oxc_linter/src/tsgolint.rs:569:8
    |
567 | impl Message<'_> {
    | ---------------- associated function in this implementation
568 |     /// Converts a `TsGoLintDiagnostic` into a `Message` with possible fixes.
569 |     fn from_tsgo_lint_diagnostic(val: TsGoLintDiagnostic, source_text: &str) -> Self {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```
